app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -0,0 +1,83 @@
+const fields = {
+  metadataKey: {
+    id: 'metadata-column-header',
+    name: 'metadata-column-header',
+    validation: [ {
+      validator: isNotEmpty,
+      message: 'Enter a column header'
+    } ]
+  },
+  metadataValue: {
+    id: 'metadata-cell-value',
+    name: 'metadata-cell-value',
+    validation: [ {
+      validator: isNotEmpty,
+      message: 'Enter a cell value'
+    } ]
+  }
+}
+
+const submissionErrorCodes = {
+  DUPLICATE_METADATA_KEYS: {
+    field: fields.metadataKey,
+    href: `#${fields.metadataKey.id}`,
+    text: 'Column header must be unique for this payment link'
+  }
+}
+
```

Comment:
I know it's unlikely but in theory we could have the 'NON_EXISTENT_METADATA_KEY' on edit submission

Reply:
As discussed, thinking through scenarios that the server could reply with is great -- I think this one is covered by: 

* if the key doesn't exist when the user clicks edit the page to edit the key will `404`

This just leaves the scenarios that the key was deleted by another user while the user was editing it, in this case there's no guidance that we can give as we don't know what happened, hopefully this is edge case enough to not model it here.